Services/Statistics/SimulationStatisticsAgent.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Clustering;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Storage;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Statistics
+{
+    public interface ISimulationStatistics
+    {
+        Task<SimulationStatisticsModel> GetSimulationStatisticsAsync(string simulationId);
+        Task CreateOrUpdateAsync(string simulationId, SimulationStatisticsModel statistics);
+    }
+
+    public class SimulationStatisticsAgent : ISimulationStatistics
```

Comment:
I'm not sure that this should be named 'Agent'. This isn't on it's own thread, is it? The other classes that have 'Agent' in their names are on their own threads, looping, interacting with actors. The use of 'Agent' here feels odd to me.